crates/polars-plan/src/dsl/function_expr/pow.rs

```diff
@@ -56,34 +85,103 @@ where
                 .into_series(),
         };
         Ok(Some(s))
-    } else if (base.len() == 1) && (exponent.len() != 1) {
-        let base = base
-            .get(0)
-            .ok_or_else(|| polars_err!(ComputeError: "base is null"))?;
+    } else {
+        pow_on_chunked_arrays(base, exponent)
+    }
+}
 
-        Ok(Some(
-            exponent
-                .apply_values(|exp| Pow::pow(base, exp))
+fn pow_to_uint_dtype<T, F>(
+    base: &ChunkedArray<T>,
+    exponent: &ChunkedArray<F>,
+) -> PolarsResult<Option<Series>>
+where
+    T: PolarsIntegerType,
+    F: PolarsIntegerType,
+    T::Native: num::pow::Pow<F::Native, Output = T::Native> + ToPrimitive,
+    ChunkedArray<T>: IntoSeries,
+{
+    let dtype = T::get_dtype();
+
+    if exponent.len() == 1 {
+        let Some(exponent_value) = exponent.get(0) else {
+            return Ok(Some(Series::full_null(base.name(), base.len(), &dtype)));
+        };
+        let s = match exponent_value.to_u64().unwrap() {
+            1 => base.clone().into_series(),
+            2..=10 => {
+                let mut out = base.clone();
+
+                for _ in 1..exponent_value.to_u8().unwrap() {
+                    out = out * base.clone()
+                }
+                out.into_series()
+            },
+            _ => base
+                .apply_values(|v| Pow::pow(v, exponent_value))
                 .into_series(),
-        ))
+        };
+        Ok(Some(s))
     } else {
-        Ok(Some(
-            polars_core::chunked_array::ops::arity::binary(base, exponent, pow_kernel)
-                .into_series(),
-        ))
+        pow_on_chunked_arrays(base, exponent)
     }
 }
 
 fn pow_on_series(base: &Series, exponent: &Series) -> PolarsResult<Option<Series>> {
     use DataType::*;
-    match base.dtype() {
-        Float32 => {
+    match (base.dtype(), exponent.dtype()) {
+        #[cfg(feature = "dtype-u8")]
+        (UInt8, UInt8 | UInt16 | UInt32 | UInt64) => {
+            let ca = base.u8().unwrap();
+            let exponent = exponent.strict_cast(&DataType::UInt32)?;
+            pow_to_uint_dtype(ca, exponent.u32().unwrap())
+        },
+        #[cfg(feature = "dtype-i8")]
+        (Int8, UInt8 | UInt16 | UInt32 | UInt64) => {
```

Comment:
hi @ritchie46 - I think you'd mentioned there is a more performant way to do this than strict-casting the exponent to u32

It didn't "register" with me though, but I also didn't press for clarification as you were preparing your presentation
Any chance you could point me to the preferred way of doing this please? Sorry to be asking again, and thanks :pray: